src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import {
   CardView,
   Container,
   DataTable,
   Spinner,
+  Toast,
 } from '@edx/paragon';
 import { useIntl } from '@edx/frontend-platform/i18n';
+import { useSearchParams } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 
 import SubHeader from '../generic/sub-header/SubHeader';
 import getPageHeadTitle from '../generic/utils';
 import messages from './messages';
 import TaxonomyCard from './taxonomy-card';
-import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded } from './data/apiHooks';
+import { useTaxonomyListDataResponse, useIsTaxonomyListDataLoaded, useDeleteTaxonomy } from './data/apiHooks';
 
 const TaxonomyListPage = () => {
   const intl = useIntl();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [showToast, setShowToast] = useState(false);
+  const [toastText, setToastText] = useState('');
+  const deleteMutation = useDeleteTaxonomy();
+
+  const onDeleteTaxonomy = (id, name) => {
+    setToastText(intl.formatMessage(messages.taxonomyDeleteToast, { name }));
+    deleteMutation.mutate({ pk: id });
+  };
+
   const useTaxonomyListData = () => {
     const taxonomyListData = useTaxonomyListDataResponse();
     const isLoaded = useIsTaxonomyListDataLoaded();
+    if (isLoaded) {
+      for (let i = 0; i < taxonomyListData.results.length; i++) {
+        taxonomyListData.results[i].onDeleteTaxonomy = onDeleteTaxonomy;
+      }
+    }
     return { taxonomyListData, isLoaded };
   };
-
   const { taxonomyListData, isLoaded } = useTaxonomyListData();
 
+  // Run only on the first render
+  useEffect(() => {
+    const deleted = searchParams.get('deleted');
+    if (deleted !== null) {
+      setToastText(intl.formatMessage(messages.taxonomyDeleteToast, { name: deleted }));
+      setShowToast(true);
+      searchParams.delete('deleted');
+      setSearchParams(searchParams);
+    }
+  }, []);
```

Comment:
If I understand correctly, this is temporarily using the search query to store the item to be deleted. I'm not sure why this is needed. Why not just store this in react state?

Reply:
I added a [comment](https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/0c6f63b2f7c921a737e51796e05020e579c5f990#diff-60367d61e5f2750d7f4224e061551f5ecf20b56aaddd1ca23ae2852a17f1e424R38) on this code. You can delete taxonomies from the Taxonomy details page, so after delete, [is redirected to the list page](https://github.com/openedx/frontend-app-course-authoring/pull/684/files#diff-0f7f5fd0f64998227e62110e00c3a7d7a5fa3da853b3689aa7e4f2fa146f37b4R40) and show the toast. Using the `searchParams` is the simplest way to show the toast in the redirect that I found.

Reply:
If you use search params, you need to make sure to use the router to reset the search params after the toast is displayed. Otherwise, if users bookmark that URL or refresh the page, they'll see the toast again. (This may sound unlikely but it happens to me all the time in Studio with the similar query param that focuses on the unit name field to let you rename a newly-created unit; it happens to me many times when the unit already exists. Very annoying.)

You could also just use localStorage to indicate that a toast should be shown though? Probably easier. Could also as on #wg-frontend if there is a standard pattern for this in our MFEs. Maybe a helper for these sort of [flash messages](https://docs.djangoproject.com/en/4.2/ref/contrib/messages/) already exists in openedx/frontend-platform or similar.

Reply:
@bradenmacdonald Thanks! I used localStorage. I didn't find a standard way, but I still left my question in #wg-frontend

Reply:
When dealing with notifications on other projects, I used this: https://notistack.com/features/basic

If we're not able to import the library, we could use the same [concept](https://github.com/iamhosseindhv/notistack/blob/master/src/SnackbarProvider/SnackbarProvider.tsx): a notification provider/context on top of our `<App />` (or `<TaxonomyLayout>` if we don't want affect other parts of the framework).

Reply:
I would prefer to avoid adding dependencies if at all possible; the bundle size is already very large.

Reply:
Local storage is good, but I'm wondering why this needs even that. Perhaps I am mistaken about this, but there is not hard refresh when navigating here, so it should be entirely possible to store this in react or redux state right?

I think these PRs are avoiding redux but even with react you can use a shared context. i.e. the root of the page can be wrapped in a <TaxonomyContextProvider> which can be a component that provides common context that all components might need. See an example [here](https://github.com/openedx/frontend-app-discussions/blob/a479f5ae5bf786e58d7a2c5c2bafc133308abfe1/src/discussions/discussions-home/DiscussionsHome.jsx#L64). 

So this context provider can be provided with the current deleted taxonomy state, i.e. what you're doing with localstorage right now. 

Here is some sample code:

Context component
```JSX
export const TaxonomyContext = React.createContext({
  deletedTaxonomy: null,
  setDeletedTaxonomy: null,
})
```
New root page for all taxonomy pages:
``JSX
function TaxonomyRoutes(){
   const [deletedTaxonomy,  setDeletedTaxonomy] = useState(null);
  return (
 <TaxonomyContext.Provider value={{deletedTaxonomy,  setDeletedTaxonomy}}>
... taxonomy routes
</TaxonomyContext.Provider>
)
}
```

Using the context:
```JSX
 const {deletedTaxonomy} = useContext(TaxonomyContext);
```

With this approach you can pass around data between components, and it may be useful in the future as well.

Reply:
I also need the same functionality to show a toast after importing a taxonomy.

I suggest we make this more generic, like:
```
   const [toastMessages,  setToastMessages] = useState([]);
```

We have the [`TaxonomyLayout`](https://github.com/openedx/frontend-app-course-authoring/blob/master/src/taxonomy/TaxonomyLayout.jsx) that is a "root" for the taxonomy pages and can be responsible to render toast/alerts.

Reply:
@xitij2000 Updated here https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/18a224a796d29d1d8154bf6f8e4ddc6f5114abba. Thanks for the feedback, that change will be help in the future. It's ready for another review round